### PR TITLE
refactor(cli,ext,ops): cleanup `regex` with `lazy-regex`

### PR DESCRIPTION
**Pull request by attila-lin**
+ bump deps: the newest `lazy-regex` need newer `oncecell` and `regex`(not sure if it matters?)
+ reduce `unwarp`
+ remove dep `lazy_static`
+ make more regex cached

### CODE REVIEW

File: cli/lsp/analysis.rs [hunk 1]
```diff
@@ -60,7 +60,7 @@ static PREFERRED_FIXES: Lazy<HashMap<&'static str, (u32, bool)>> =
   });
 
 static IMPORT_SPECIFIER_RE: Lazy<Regex> =
-  Lazy::new(|| Regex::new(r#"\sfrom\s+["']([^"']*)["']"#).unwrap());
+  lazy_regex::lazy_regex!(r#"\sfrom\s+["']([^"']*)["']"#);
```
Comment (bartlomieju) [inline cli/lsp/analysis.rs:63]: What's the advantage of using `lazy_regex` over current solution?

Comment (dsherret) [reply cli/lsp/analysis.rs:63]: I think we should keep using once_cell only because it's an easy solution and doesn't involve adding another crate.

Comment (bartlomieju) [reply cli/lsp/analysis.rs:63]: I tend to agree

Comment (attila-lin) [reply cli/lsp/analysis.rs:63]: `lazy_regex` just provide some macro which is widely used(65k+ downloaded) and small.

The macro kind solution is also recommended by [once_cell](https://docs.rs/once_cell/latest/once_cell/#lazily-compiled-regex).

Comment (attila-lin) [reply cli/lsp/analysis.rs:63]: About the advantages, it

+ provide same features with less code.
+ no unwrap.
+ At most cases, we should create a cached regex, other code contributors can follow the specification.

Comment (dsherret) [reply cli/lsp/analysis.rs:63]: It does look nice, but it’s yet another crate that could go stale or get out of sync with our shared dependencies and using a lazy is not a big deal.

We should remove the usages of lazy_static and cache that one regex though.

Comment (crowlKats) [reply cli/lsp/analysis.rs:63]: I agree; it doesnt seem necessary for no real benefit

Comment (attila-lin) [reply cli/lsp/analysis.rs:63]: ok, I will fix it

**Comment by attila-lin** [comment]
Yes of course

**Comment by bartlomieju** [comment]
@attila-lin, I'm still interested in landing this PR, seems like a good change. Any chance you could rebase it?

**Comment by attila-lin** [comment]
Be busy last month😅. 
Working on this today~

**Comment by attila-lin** [comment]
> @attila-lin, I'm still interested in landing this PR, seems like a good change. Any chance you could rebase it?

hi, I did the rebase.

**Comment by bartlomieju** [comment]
Thanks for looking into this @attila-lin. From what I can see in changed lines it feels like _all_ of these regex should be lazy statics - there's really no point in having them defined and compile on each invocation of the functions they are defined in. 

It would be great if you could lift all of them to the top level of the module as define them as `Lazy<Regex>`.

**Review by bartlomieju — APPROVED** [review]
Disregard previous comment, I misunderstood how the `lazy_regex::regex` works. LGTM, thanks @attila-lin

**Pull request merged**